src/app/components/generics/property-input/text-input.component.ts

```diff
@@ -16,25 +19,79 @@ export class TextInputComponent implements OnInit {
   @Input() maxLines = 1;
   @Input() isLink: boolean;
   @Input() pattern?: string;
+  @Input() isLatexInput: boolean;
 
   inputValue: string;
+  packedLatexValue: string;
+  renderOutput = 'pdf';
+  urlToRenderedPdfBlob: string;
+
+  constructor(
+    private utilService: UtilService,
+    private latexRendererService: RenderLatexControllerService
+  ) {}
 
   saveChanges(): void {
-    this.onSaveChanges.emit(this.inputValue);
+    if (this.isLatexInput) {
+      this.onSaveChanges.emit(this.packedLatexValue);
+    } else {
+      this.onSaveChanges.emit(this.inputValue);
+    }
   }
 
   inputChanged(): void {
-    if (!this.inputValue) {
-      this.inputValue = null;
+    if (this.isLatexInput) {
+      this.onChange.emit(this.packedLatexValue);
+    } else {
+      if (!this.inputValue) {
+        this.inputValue = null;
+      }
+      this.onChange.emit(this.inputValue);
     }
-    this.onChange.emit(this.inputValue);
   }
 
   openLink(): void {
     window.open(this.inputValue, '_blank');
   }
 
-  ngOnInit() {
-    this.inputValue = this.value;
+  ngOnInit(): void {
+    if (this.isLatexInput) {
+      this.packedLatexValue = this.value;
+      if (this.value) {
+        this.renderLatexContent(this.packedLatexValue);
+      }
+    } else {
+      this.inputValue = this.value;
+    }
+  }
+
+  openLatexEditor(): void {
+    const dialogRef = this.utilService.createDialog(
+      LatexEditorDialogComponent,
+      {
+        title: 'LaTeX Render Editor',
+        packedLatexValue: this.packedLatexValue,
+      },
+      'auto'
+    );
+
+    dialogRef.afterClosed().subscribe((dialogResult) => {
+      if (
+        dialogResult !== undefined &&
```

Comment:
use `!= null` instead.

Reply:
dialogResult equals undefined if dialogRef.close() is called without any argument, which is the case if one clicks outside the dialog and it gets closed. In that case, the save button gets displayed without any changes made to the field.